components/type_info.rst

```diff
@@ -0,0 +1,59 @@
+The TypeInfo Component
+======================
+
+    The TypeInfo component extracts PHP types information. It aims to:
+
+    - Have a powerful Type definition that can handle union, intersections, and generics (and could be even more extended)
+
+    - Being able to get types from anything, such as properties, method arguments, return types, and raw strings (and can also be extended).
+
+
```

Comment:
We need to mention that this component was introduced as an experimental feature. Ideally, with a link to https://symfony.com/doc/current/contributing/code/experimental.html.